doc/stages/filters.hag.rst

```diff
@@ -164,7 +164,11 @@ max_distance
     option.  [Default: N/A]
 
 allow_extrapolation
-    Does no extrapolation.  For ``HeightAboveGround`` to be calculated,
-    a ground point must be at the same location as the non-ground point
-    being evaluated.  Not used with the `delaunay`_ option.
+    If false and a non-ground point lies outside of the bounding box of
+    all ground points, its ``HeightAboveGround`` is set to 0.  If true
+    and ``delaunay`` is set, the ``HeightAboveGround`` is set to the
+    difference between the heights of the non-ground point and nearest
+    ground point.  If true and using neighbor interpolation (``delaunay``
+    is false), extrapolation is used to assign the ``HeightAboveGround``
+    value.
```

Comment:
> extrapolation is used to assign the ``HeightAboveGround`` value.

Still a little confusing, as it is all a form of extrapolation, regardless of if you are using Delaunay or neighbor interpolation. As I read it, the neighbor interpolation just continues to do what it does inside the bounding box, which is effectively IDW. I need to reread, but I'm not sure we ever even say this. IIRC, we call it a "weighted" approach rather than using the common name.

Reply:
Not exactly.  For delaunay interpolation there is no extrapolation, really.  If the switch is set to true and the test point is outside of the triangulation, we just use the height of the closest ground point as the base.

I get that this is confusing, which is why I wondered if we needed it at all.  I'm unsure what to do.